api/python/provisioner/commands/__init__.py

```diff
@@ -685,6 +711,71 @@ def run(self, targets):  # noqa: C901 FIXME
             ) from update_exc
 
 
+@attr.s(auto_attribs=True)
+class RollbackUpdate(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsRollbackUpdate
+
+    def run(self, target_version, targets):
+
+        local_minion = local_minion_id()
+        rollback_path = PillarKey('rollback')
+        rollback_pillar = PillarResolver(local_minion).get([rollback_path])
+        rollback_pillar = rollback_pillar[local_minion][rollback_path]
+        cortx_tgt_version = (
+            target_version
+            if target_version in rollback_pillar.keys()
+            else None
+        )
+
+        if cortx_tgt_version is None:
+            logger.exception(
+                f'{target_version} version data not available for rollback'
+            )
+        else:
+            txn_id_path = PillarKey(f"rollback/{cortx_tgt_version}")
+            txn_id_pillar = PillarResolver(local_minion).get([txn_id_path])
+            txn_id_pillar = next(iter(txn_id_pillar.values()))
+            try:
+                for target in txn_id_pillar[txn_id_path]:
+                    txn_id = txn_id_pillar[txn_id_path][target]['yum_txn_id']
+
+                    if not txn_id or txn_id is values.MISSED:
+                        raise BadPillarDataError(
+                            f"yum txn id not available for {target}"
+                        )
+                    else:
+                        logger.info(f"Starting rollback on target {target}")
+                        salt_cmd_run(
+                            f"yum history rollback -y {txn_id}",
+                            targets=target
+                        )
+                        logger.info(
+                            f"Rollback on target {target} is completed"
+                        )
+
+                    logger.info('Restoring configurations for components')
+                    for component in (
+                        'motr',
+                        's3server',
+                        'hare',
+                        'ha.cortx-ha',
+                        'sspl',
+                        'uds',
+                        'csm'
+                    ):
+                        _rollback_component(component, targets)
+                    logger.info(
+                        "Configurations restored successfully "
+                        f"on target {target}"
+                    )
+            except Exception as exc:
+                logger.exception(
+                    'failed to rollback sw stack, reason {}'
+                    .format(exc)
+                )
```

Comment:
please add `raise` here to propagate or raise some specific exception

Reply:
addressed